packages/scoutgame/src/builderNfts/recordNftTransfer.ts

```diff
@@ -0,0 +1,78 @@
+import { prisma } from '@charmverse/core/prisma-client';
+import type { Address } from 'viem';
+
+import { refreshScoutNftBalance } from './refreshScoutNftBalance';
+
+type RecordNftTransferParams = {
+  from: Address;
+  to: Address;
+  tokenId: number;
+  amount: number;
+  contractAddress: Address;
+  txHash: string;
+};
+
+export async function recordNftTransfer({
+  contractAddress,
+  amount,
+  from,
+  to,
+  tokenId,
+  txHash
+}: RecordNftTransferParams): Promise<void> {
+  const matchingNft = await prisma.builderNft.findFirstOrThrow({
+    where: {
+      tokenId,
+      contractAddress: {
+        equals: contractAddress,
+        mode: 'insensitive'
+      }
+    }
+  });
+
+  await Promise.all([
+    refreshScoutNftBalance({
+      wallet: from,
+      tokenId,
+      contractAddress,
+      nftType: 'default'
+    }),
+    refreshScoutNftBalance({
+      wallet: to,
+      tokenId,
+      contractAddress,
+      nftType: 'default'
+    })
+  ]);
+
+  const existingNftPurchaseEvent = await prisma.nFTPurchaseEvent.findFirst({
+    where: {
+      builderNftId: matchingNft.id,
+      txHash,
```

Comment:
isnt txHash enough to look for an existing event? if there was an nft even w/the same txhash but different props, should that log an error?